cardano-cli/src/Cardano/CLI/Run/Hash.hs

```diff
@@ -118,6 +121,29 @@ runHashAnchorDataCmd Cmd.HashAnchorDataCmdArgs{toHash, mExpectedHash, mOutFile}
         mkHandler :: (Monad m, Exception e) => (e -> HttpRequestError) -> Handler m HashCmdError
         mkHandler x = Handler $ return . HashGetFileFromHttpError . x
 
+    convertToHttp :: URI -> ExceptT HashCmdError IO URI
+    convertToHttp ipfsUri = do
+      mIpfsGatewayUriString <- handleIOExceptT HashReadEnvVarError $ IO.lookupEnv "IPFS_GATEWAY_URI"
+      ipfsGatewayUriString <- expectJustExceptT HashIpfsGatewayNotSetError mIpfsGatewayUriString
+      ipfsGatewayUri <-
+        expectJustExceptT (HashInvalidURLError ipfsGatewayUriString) $ parseAbsoluteURI ipfsGatewayUriString
+      return $
+        ipfsGatewayUri
+          { uriPath =
+              '/'
+                : intercalate
+                  "/"
+                  ( pathSegments ipfsGatewayUri
+                      ++ ["ipfs"]
+                      ++ maybe [] (\ipfsAuthority -> [uriRegName ipfsAuthority]) (uriAuthority ipfsUri)
+                      ++ pathSegments ipfsUri
+                  )
+          }
+
+  expectJustExceptT :: e -> Maybe a -> ExceptT e IO a
```

Comment:
There is  https://hackage.haskell.org/package/transformers-except-0.1.4/docs/Control-Monad-Trans-Except-Extra.html#v:hoistMaybe

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/pull/895/commits/fe49f9b37e43a0bfa070802421b37b891f0bea73